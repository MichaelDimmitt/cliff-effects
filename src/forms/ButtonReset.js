import React from 'react';

// PROJECT COMPONENTS
import { BigButton } from './inputs';


const BLANK_CALLBACK = function () {};

const ButtonReset = function ({ children, onClick, overrides }) {

  if (!overrides) {
    overrides = {};
  };

  let clickHandler = function () {
    onClick({
      header:    ``,
      message:   `default`,
      leaveText: `Start New Client`,
      callback:  { BLANK_CALLBACK },
    });
  };

  return (
    <BigButton
      className = { `caution` }
      onClick   = { clickHandler }
      overrides = { overrides }>
      { children }
    </BigButton>
  );

};  // Ends <ButtonReset>


export { ButtonReset };
