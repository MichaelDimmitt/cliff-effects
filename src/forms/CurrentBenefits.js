// REACT COMPONENTS
import React from 'react';
import cloneDeep from 'lodash/cloneDeep';

// PROJECT COMPONENTS
import { FormPartsContainer } from './FormPartsContainer';
import { ControlledRadioYesNo } from './inputs';

// DATA
import { allBenefitOrders } from '../programs/allBenefitOrders';


/** Asks which benefits the user is currently receiving
 *
 * @todo Add 'vertical list of options' creator that will create a list of fields using the `.field-aligner` class
 *
 * @function
 * @param {object} props
 * @property {object} props.current Client current info.
 * @property {function} props.updateClientValue Updates state upstream.
 * @property {function} props.translations Uses user chosen language-specific
 *    translations.
 *
 * @returns {object} Component
 */
class CurrentBenefitsContent extends React.Component {
  handleRadioChange = (event, inputProps) => {
    const benefitName = inputProps.name,
          route       = `benefits`,
          value       = cloneDeep(this.props.current.benefits);

    if (inputProps.value) {
      value.push(benefitName);
      value.sort(
        // Make sure benefits are in the correct order
        (a, b) => {
          // `props.benefits` is list of available benefits
          const aIndex = this.props.benefits.indexOf(a),
                bIndex = this.props.benefits.indexOf(b);

          return aIndex - bIndex;
        }
      );
    } else {
      const index = value.indexOf(benefitName);

      if (index >= 0) {
        value.splice(index, 1);
      }
    }  // ends if input value

    this.props.updateClientValue(event, { route, value, time: `current` });
  };  // Ends handleRadioChange()

  render() {
    const {
      current,
      translations,
      benefits,
    } = this.props;

    const components = [];
  
    for (let benefitIndex = 0; benefitIndex < benefits.length; benefitIndex++) {
      const benefit        = benefits[ benefitIndex ],
            translationKey = `i_has_${benefit}_label`;

      // @todo This would have to be handled in every translation situation.
      // @todo Also, is there a way we can include this in the translation report?
      if (!(translationKey in translations)) {
        throw new Error(`No translation found as label for benefit radio buttons; expected key "${translationKey}"`);
      }

      const labelText = translations[ translationKey ];
  
      components.push(
        <ControlledRadioYesNo
          key          = { benefit }
          translations = { translations }
          labelText    = { labelText }
          onChange     = { this.handleRadioChange }
          checked      = { current.benefits.includes(benefit) }
          name         = { benefit } />
      );
    }  // ends for every benefit

    return (
      <div>{ components }</div>
    );
  };  // Ends render()
};  // Ends <CurrentBenefitsContent>

/** 
 * @function
 * @param {object} props
 * @property {function} props.updateClientValue Updates state upstream.
 * @property {object} props.navData Bottom row buttons
 * @property {object} props.client JSON object with future and current values.
 * @property {function} props.translations Uses user chosen language-specific
 *    translations.
 *
 * @returns {object} Component
 */
const CurrentBenefitsStep = ({ updateClientValue, navData, client, translations }) => {
  // @todo Combine with related components since there are so few?

  return (
    <FormPartsContainer
      title     = { translations.i_currentBenefits }
      clarifier = { translations.i_selectBenefits }
      navData   = { navData }
      formClass = { `benefits` }
      formSize  = { `massive` }>
      <CurrentBenefitsContent
        updateClientValue = { updateClientValue }
        current           = { client.current }
        translations      = { translations }
        benefits          = { allBenefitOrders[ client.USState ] } />
    </FormPartsContainer>
  );

};  // Ends <CurrentBenefitsStep>


export { CurrentBenefitsStep };
