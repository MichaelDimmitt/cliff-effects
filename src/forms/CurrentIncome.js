// REACT COMPONENTS
import React from 'react';
import { Form } from 'semantic-ui-react';

// PROJECT COMPONENTS
import { FormPartsContainer } from './FormPartsContainer';
import { IntervalColumnHeadings } from '../components/headings';
import { CashFlowInputsRow } from './cashflow';


// ========================================
// COMPONENTS
// ========================================
/* Move to program calculations:
 * @todo Figure out which programs need to know which types of incomes
 * and categorize/tag them accordingly.
 * @todo Calc and store `client.currentUnearnedIncomeMonthly`? I think
 * we do still have to keep the other specific income soruces separate
 * as they're possibly used in other calculations.
 * @todo Stuff like interest of bank accounts? (unearned income?)
 * @todo Other assets (not counted in gross income? income categories?)
 * @todo Relevant? "State housing programs base eligibility on net yearly income.
 * Net yearly income does not include funds such as wages earned by full-time
 * students, worker's compensation, and a certain amount of wages earned by a
 * tenant 62 or older. It also allows you to deduct certain amounts, such as
 * necessary medical expenses and personal care services." (@see {@link
 * http://www.masslegalhelp.org/housing/financial-eligibility})
 */

/**
 * @todo Add note: "Household income (a before tax income, and does not include
 * funds such as income from children under 18 years old, amounts received
 * through training programs funded by HUD, and the income of a live-in aide)"
 * (@see {@link http://www.masslegalhelp.org/housing/financial-eligibility})
 */

/** Contents of income step. Abstracted to allow entry of `future` values too.
 *
 * @function
 * @param {object} props
 * @property {object} props.current Client current info. Could be
 *     changed to just 'client' to allow future values in abstraction.
 * @property {string} props.time 'current' or 'future'. (needed?)
 * @property {function} props.updateClientValue Updates state upstream.
 * @property {function} props.translations Uses user chosen language-specific text.
 *
 * @returns {object} React element
 */
const IncomeForm = function ({ current, time, updateClientValue, translations }) {

  let type = `income`;

  /** Makes sure values are propagated to 'future' properties if needed
   * @member
   * @depricated
   */
  let ensureFuture = function (evnt, inputProps) {
    updateClientValue(evnt, { ...inputProps, fillFuture: true });
  };

  let sharedProps = {
    timeState:         current,
    time:              time,
    type:              type,
    updateClientValue: ensureFuture,
  };

  return (
    <div className={ `field-aligner two-column` }>

      <IntervalColumnHeadings type={ type } />

      {/* Need to give user more info on these. */}
      
      <CashFlowInputsRow
        { ...sharedProps }
        generic = { `earned` }>
        { translations.earnedIncome.i_label }
      </CashFlowInputsRow>
      <CashFlowInputsRow
        { ...sharedProps }
        generic = { `TAFDC` }> 
        { translations.TAFDC.i_label }
      </CashFlowInputsRow>
      <CashFlowInputsRow
        { ...sharedProps }
        generic = { `SSI` }> 
        { translations.SSI.i_label }
      </CashFlowInputsRow>
      <CashFlowInputsRow
        { ...sharedProps }
        generic = { `SSDI` }>
        { translations.SSDI.i_label }
      </CashFlowInputsRow>
      <CashFlowInputsRow
        { ...sharedProps }
        generic = { `childSupportIn` }>
        { translations.childSupport.i_label }
      </CashFlowInputsRow>
      <CashFlowInputsRow
        { ...sharedProps }
        generic = { `unemployment` }> 
        { translations.unemployment.i_label }      
      </CashFlowInputsRow>
      <CashFlowInputsRow
        { ...sharedProps }
        generic = { `workersComp` }> 
        { translations.workersComp.i_label }
      </CashFlowInputsRow>
      <CashFlowInputsRow
        { ...sharedProps }
        generic = { `pension` }>
        { translations.pension.i_label }
      </CashFlowInputsRow>
      <CashFlowInputsRow
        { ...sharedProps }
        generic = { `socialSecurity` }>
        { translations.socialSecurity.i_label }
      </CashFlowInputsRow>
      <CashFlowInputsRow
        { ...sharedProps }
        generic = { `alimony` }> 
        { translations.alimony.i_label }
      </CashFlowInputsRow>
      <CashFlowInputsRow
        { ...sharedProps }
        generic = { `otherIncome` }>
        { translations.otherIncome.i_label }
      </CashFlowInputsRow>
      <Form.Field>{ translations.i_explainSnapCalculation }</Form.Field>

    </div>
  );  // ends return

};  // Ends <IncomeForm>


/**
 * @function
 * @param {object} props
 * @property {function} props.updateClientValue Updates state upstream.
 * @property {object} props.navData Bottom row buttons. 
 * @property {object} props.client JSON object with `future` and `current` props.
 * @property {function} props.translations Uses user chosen language-specific text.
 *
 * @returns {object} React element
 */
const CurrentIncomeStep = function ({ updateClientValue, navData, client, translations }) {
// `props` is a cloned version of the original props. References broken.

  return (
    <FormPartsContainer
      title     = { translations.i_title }
      clarifier = { translations.i_clarifier }
      navData   = { navData }
      formClass = { `income` }>
      <IncomeForm
        updateClientValue = { updateClientValue }
        current           = { client.current }
        time              = { 'current' }
        translations      = { translations } />
    </FormPartsContainer>
  );

};


export {
  CurrentIncomeStep,
  IncomeForm,
};
