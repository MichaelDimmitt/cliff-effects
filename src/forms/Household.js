// REACT COMPONENTS
import React from 'react';
import {
  Button,
  Form,
  Header,
  Checkbox,
  Icon,
} from 'semantic-ui-react';

import Select from 'react-select';

// PROJECT COMPONENTS
import { FormPartsContainer } from './FormPartsContainer';
import { ColumnHeading } from '../components/headings';
import { ManagedNumberField } from './inputs';

// COMPONENT HELPER FUNCTIONS
import {
  isNonNegWholeNumber,
  hasOnlyNonNegWholeNumberChars,
} from '../utils/validators';

// OBJECT MANIPULATION
import { cloneDeep } from 'lodash';


// ======================
// REUSED
// ======================
const Columns = {};

// `noMargin` is a bit hacky, but it'll do for now
Columns.One = function ({ children, className }) {
  className = className || ``;
  return (
    <div className={ className + ` column column-one` }>
      { children }
    </div>
  );
};

Columns.Two = function ({ children, className }) {
  className = className || ``;
  return (
    <div className={ className + ` column column-two` }>
      { children }
    </div>
  );
};

Columns.Three = function ({ children, className }) {
  className = className || ``;
  return (
    <div className={ className + ` column column-three` }>
      { children }
    </div>
  );
};

Columns.Four = function ({ children, className }) {
  className = className || ``;
  return (
    <div className={ className + ` column column-four` }>
      { children }
    </div>
  );
};


const ColumnHeader = function ({ children, columnNum }) {

  let Container = Columns[ columnNum ];

  return (
    <Container>
      <ColumnHeading
        type    = { `household` }
        colName = { `` }>
        { children }
      </ColumnHeading>
    </Container>
  );

};


const MemberButton = function ({ basic, color, iconName, className, onClick }) {

  color = color || null;

  return (
    <Button
      circular
      type      = { `button` }
      basic     = { !!basic }
      color     = { color }
      icon      = { iconName }
      className = { className }
      onClick   = { onClick } />
  );

};


// ======================
// UNIQUE
// ======================

const Role = function ({ member, setMember, setDropdownMember, translations }) {

  let thisRole  = null,
      className = `head`;

  if (member.index === 0) {

    thisRole = (<span>{ translations.i_headOfHousehold }</span>);

  } else if (member.index === 1) {

    className = `second-member-choice`;  // -1em

    let options = [
      { label: translations.i_spouse, value: 'spouse', name: 'm_role' },
      { label: translations.i_childOther, value: 'member', name: 'm_role' },
    ];

    let selectedValue = options.find((obj) => {
      return obj.value === member.m_role;
    });

    thisRole = (
      <Select
        selection
        name     = { `m_role` }
        value    = { selectedValue }
        options  = { options }
        onChange = { setDropdownMember } />
    );

  } else {

    thisRole = (<span>{ translations.i_childOther }</span>);

  }  // ends which member index

  // Styles will have to be adjusted.
  return (<div className={ `role ` + className }>{ thisRole }</div>);

};  // Ends <Role>


const MemberField = function ({ household, time, setHousehold, updateClientValue, translations }, indx) {

  let member     = household[ indx ],
      routeStart = `household/` + indx + `/`;
  member.index   = indx;  // Just needed as member prop in this file

  let onMemberChange = function (evnt, inputProps) {
    let route = routeStart + inputProps.name,
        data  = { route: route, value: inputProps.value };
    updateClientValue(evnt, data);
  };

  let onMemberDropdownChange = function (selectedOption) {
    const data = {
      route: routeStart + selectedOption.name,
      value: selectedOption.value,
    };
    updateClientValue(selectedOption, data);
  };


  let onMemberChecked = function (evnt, inputProps) {
    let route = routeStart + inputProps.name,
        data  = { route: route, value: inputProps.checked };
    updateClientValue(evnt, data);
  };

  let removeMember = function (evnt, inputProps) {
    household.splice(indx, 1);
    setHousehold(evnt, household);
  };

  // For keyboard access (already does spacebar)
  let onKeyDown = function (evnt) {
    if (evnt.key === `Enter`) {
      evnt.target.click();
    }
  };

  let format = function (value) {
    return value;
  };

  let onBlur = function () {
    return true;
  };

  // The font size thing is a bit weird, but... later
  return (
    <Form.Field
      className = { `flex-item member` }
      key       = { indx }>

      <Columns.One>
        { (indx > 0) ? (
          <MemberButton
            className = { `remove` }
            onClick   = { removeMember }
            iconName  = { `remove` } />
        ) : (
          <span>
            { (household.length > 1) ? (
              <Icon
                fitted
                className = { `no-removing` }
                name      = { `ban` } />
            ) : (
              null
            ) }
          </span>
        ) }
      </Columns.One>

      <Columns.Two>
        <Role
          member           = { member }
          setMember        = { onMemberChange }
          setDropdownMember={ onMemberDropdownChange }
          translations     = { translations } />
      </Columns.Two>

      <Columns.Three>
        <ManagedNumberField
          value            = { member.m_age }
          name             = { `m_age` }
          className        = { time + ` member-age ` }
          displayValidator = { hasOnlyNonNegWholeNumberChars }
          storeValidator   = { isNonNegWholeNumber }
          format           = { format }
          store            = { onMemberChange }
          onBlur           = { onBlur } />
      </Columns.Three>

      <Columns.Four>
        <Checkbox
          name      = { `m_disabled` }
          checked   = { member.m_disabled }
          onChange  = { onMemberChecked }
          onKeyDown = { onKeyDown } />
      </Columns.Four>

    </Form.Field>
  );

};  // Ends <MemberField>


const getMembers = function (current, time, setHousehold, updateClientValue, translations) {

  let household = current.household,
      props     = {
        household:         household,
        time:              time,
        setHousehold:      setHousehold,
        updateClientValue: updateClientValue,
        translations:      translations,
      };

  let mems = [];
  for (let memi = 0; memi < household.length; memi++) {
    mems.push(MemberField(props, memi));
  };

  return mems;

};


const HouseholdContent = function ({ current, time, updateClientValue, translations }) {

  // Don't mutate state properties
  let household = cloneDeep(current.household);

  let setHousehold = function (evnt, newHousehold) {

    let obj = {
      route: `household`,
      value: newHousehold,
    };

    updateClientValue(evnt, obj);

  };

  let addMember = function (evnt, inputProps) {

    let member;
    if (household.length === 1) {
      member = { m_age: 30, m_role: `spouse`, m_disabled: false };
    } else {
      member = { m_age: 12, m_role: `member`, m_disabled: false };
    }

    household.push(member);
    setHousehold(evnt, household);

  };

  return (
    <div className={ `field-aligner two-column` }>
      <div className={ `column-headers flex-item` }>
        <ColumnHeader columnNum={ `One` } />
        <ColumnHeader columnNum={ `Two` }>{ translations.i_role }</ColumnHeader>
        <ColumnHeader columnNum={ `Three` }>{ translations.i_age }</ColumnHeader>
        <ColumnHeader columnNum={ `Four` }>{ translations.i_disabled }</ColumnHeader>
      </div>

      { getMembers(current, time, setHousehold, updateClientValue, translations) }

      <Button
        basic
        type    = { `button` }
        id      = { `addMember` }
        onClick = { addMember }>
        <Columns.One className={ `add-row-column` }>
          <Icon
            circular
            inverted
            name  = { `plus` }
            color = { `teal` } />
        </Columns.One>

        <Columns.Two className={ `add-row-column` }>
          <Header
            as    = { `h4` }
            color = { `teal` }>
            { translations.i_addMember }
          </Header>
        </Columns.Two>

        <Columns.Three className={ `add-row-column` } />
        <Columns.Four className={ `add-row-column` } />
      </Button>

    </div>
  );

};  // Ends <HouseholdContent>


// `props` is a cloned version of the original props. References broken.
const HouseholdStep = function ({ updateClientValue, navData, client, translations }) {

  return (
    <FormPartsContainer
      title     = { translations.i_title }
      clarifier = { translations.i_clarifier }
      navData   = { navData }
      formClass = { `household` }>
      <HouseholdContent
        updateClientValue = { updateClientValue }
        current           = { client.current }
        time              = { `current` }
        translations      = { translations } />
    </FormPartsContainer>
  );

};


export { HouseholdStep };
