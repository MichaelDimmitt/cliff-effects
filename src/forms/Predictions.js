import React from 'react';
import {
  Divider,
  Header,
  Tab,
  Message,
  Button,
  Menu,
} from 'semantic-ui-react';

// PROJECT COMPONENTS
import { FormPartsContainer } from './FormPartsContainer';
import { IntervalColumnHeadings } from '../components/headings';
import { CashFlowInputsRow } from './cashflow';
import { GraphHolder } from './output/GraphHolder';
import { Summary } from './output/Summary';
import { BenefitsTable } from './output/BenefitsTable';
import { ResourcesColumns } from './output/ResourcesColumns';
import { StackedResources } from './output/StackedResources';
import { BenefitsLines } from './output/BenefitsLines';


/** @todo Cash flow row for trying out different future incomes.
 *
 * As per Project Hope's input, for the first prototype
 *     we're only including the ability to change earned income.
 *
 * @function
 * @param {object} props
 * @param {object} props.future Client future/predictive data.
 * @param {string} props.time Used in class names. Meant to make
 *     this more easily decoupled in future.
 * @param {function} props.updateClientValue Update client state
 *     value.
 * @param {object} props.translations Language-specific text
 *
 * @returns {object} React element
 */
const IncomeForm = function ({ future, time, updateClientValue, translations }) {

  let type = `income`;

  return (
    <div className={ `field-aligner two-column` }>
      <IntervalColumnHeadings type={ type } />
      <CashFlowInputsRow
        timeState         = { future }
        type              = { type }
        time              = { time }
        updateClientValue = { updateClientValue }
        generic           = { `earned` }
        labelInfo         = { `(Weekly pay = hourly wage times average number of work hours per week)` }>
        { translations.i_futureIncomeQuestion }
      </CashFlowInputsRow>
    </div>
  );
};


const TabbedVisualizations = ({ client, openFeedback, translations }) => {
  return (
    <Tab
      menu  = {{ color: `teal`,  attached: true, tabular: true }}
      panes = { [
        {
          menuItem: (
            <Menu.Item
              key = { `tab0` }
              as  = { Button }>
              { translations.i_summaryTitle }
            </Menu.Item>
          ),
          render: () => { return (
            <Tab.Pane><Summary
              client       = { client }
              openFeedback = { openFeedback }
              translations = { translations } />
            </Tab.Pane>
          );}, 
        },
        { 
          menuItem: (
            <Menu.Item
              key = { `tab1` }
              as  = { Button }>
              { translations.i_tabTitleChanges }
            </Menu.Item>
          ),
          render: () => {
            return (
              <Tab.Pane>
                <BenefitsTable
                  client       = { client }
                  translations = { translations } />
              </Tab.Pane>
            );
          },
        },
        {
          menuItem: (
            <Menu.Item
              key = { `tab2` }
              as  = { Button }>
              { translations.i_tabTitleChangesChart }
            </Menu.Item>
          ),
          render: () => { return (
            <Tab.Pane>
              <GraphHolder
                client       = { client }
                Graph        = { ResourcesColumns }
                translations = { translations } />
            </Tab.Pane>
          );},
        },
        {
          menuItem: (
            <Menu.Item
              key = { `tab3` }
              as  = { Button }>
              { translations.i_tabTitleStackedIncomes }
            </Menu.Item>
          ),
          render: () => {
            return (
              <Tab.Pane>
                <GraphHolder
                  client       = { client }
                  Graph        = { StackedResources }
                  translations = { translations } />
              </Tab.Pane>
            );
          },
        },
        {
          menuItem: (
            <Menu.Item
              key = { `tab4` }
              as  = { Button }>
              { translations.i_tabTitleBenefitPrograms }
            </Menu.Item>
          ),
          render: () => {
            return (
              <Tab.Pane>
                <GraphHolder
                  client       = { client }
                  Graph        = { BenefitsLines }
                  translations = { translations } />
              </Tab.Pane>
            );
          },
        },
      ] } />
  );
};  // Ends <TabbedVisualizations>


const PredictionsStep = function ({ updateClientValue, navData, client, translations, openFeedback }) {

  return (
    <FormPartsContainer
      title     = { translations.i_title }
      clarifier = { null }
      navData   = { navData }
      formClass = { `predictions` }>
      {/* `predictions-form`: This whole div will be outside
        the form in the future and then we'll be able to
        access its style that way */}
      <div id={ `predictions-form` }>
        <IncomeForm
          updateClientValue = { updateClientValue }
          future            = { client.future }
          time              = { `future` }
          translations      = { translations } />
        <Divider className={ `ui section divider hidden` } />
      </div>
      <div id={ `results-intro` }>
        <Header
          as        = { `h3` }
          className = { `ui Header align centered` }>
          { translations.i_chartsHeader }
        </Header>
        <Message
          visible
          warning
          className = { `prediction-message` }>
          { translations.i_warningMessage }
          <Button
            compact
            className = { `feedback-button` }
            size      = { `small` }
            color     = { `teal` }
            onClick   = { openFeedback }>
            { translations.i_submitFeedback }
          </Button>
        </Message>
      </div>
      <TabbedVisualizations 
        client       = { client }
        openFeedback = { openFeedback }
        translations = { translations } />
    </FormPartsContainer>
  );
};  // End <PredictionsStep>


export { PredictionsStep };
