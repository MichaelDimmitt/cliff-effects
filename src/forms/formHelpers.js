/** @module formHelpers
 * @todo Separate into better-named dirs/files
 */

// REACT COMPONENTS
import React from 'react';
import { Icon } from 'semantic-ui-react';

// ========================================
// INPUT CONTAINER COMPONENTS
// ========================================

/** Red bold text that only appears if `validRow` is `false` */
const InvalidMessage = function ({ validRow, className, children }) {
  className = className || ``;

  var result = null;
  if (!validRow && children) {
    result = (
      <div className = { `invalid ` + className }>
        { children }
      </div>
    );
  }

  return result;
};  // End <InvalidMessage>


var AttentionArrow = function () {

  return (
    <span className={ 'attention-arrow' }>
      <Icon
        className = { 'attention-font' }
        fitted
        name      = { 'angle right' }
        size      = { 'big' } />
      <Icon
        className = { 'attention-font' }
        fitted
        name      = { 'angle right' }
        size      = { 'big' } />
    </span>
  );

};  // End <AttentionArrow>


export {
  InvalidMessage,
  AttentionArrow,
};
