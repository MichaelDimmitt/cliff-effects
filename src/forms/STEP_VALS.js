import { CurrentIncomeStep } from './CurrentIncome';
import { CurrentExpensesStep } from './CurrentExpenses';
import { PredictionsStep } from './Predictions';
import { HouseholdStep } from './Household';
import { CurrentBenefitsStep } from './CurrentBenefits';

export const STEP_VALS = [
  {
    component: CurrentBenefitsStep,
    key:       'currentBenefits',
    time:      'current',
  },
  {
    component: HouseholdStep,
    key:       'household',
    time:      'current',
  },
  {
    component: CurrentIncomeStep,
    key:       'currentIncome',
    time:      'current',
  },
  {
    component: CurrentExpensesStep,
    key:       'currentExpenses',
    time:      'current',
  },
  {
    component: PredictionsStep,
    key:       'predictions',
    time:      'future',
  },
]; // end STEP_VALS
