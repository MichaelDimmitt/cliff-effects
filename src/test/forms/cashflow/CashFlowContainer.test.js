import React from 'react';
import { mount } from 'enzyme';

import { CashFlowRow } from '../../../forms/cashflow';

describe('<CashFlowRow>', () => {
  it('matches snapshot', () => {
    const child = <span>Some content</span>;

    const validProps = {
      label:    'Rent',
      validRow: true,
      children: child,
    };
    expect(mount(<CashFlowRow { ...validProps } />)).toMatchSnapshot();

    const invalidProps = {
      label:    'Rent',
      validRow: false,
      message:  'What did you enter!?',
      children: child,
    };
    expect(mount(<CashFlowRow { ...invalidProps } />)).toMatchSnapshot();
  });
});
