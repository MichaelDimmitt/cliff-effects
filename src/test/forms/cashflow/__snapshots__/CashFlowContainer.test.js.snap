// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CashFlowRow> renders with message 1`] = `
<CashFlowRow
  label="Rent"
  message="What did you enter!?"
  validRow={false}
>
  <FormField
    className="cashflow"
    inline={true}
  >
    <div
      className="inline field cashflow"
    >
      <WithMessageAbove
        isUserError={true}
        message="What did you enter!?"
      >
        <div
          className="with-message-above "
        >
          <div
            className="message-above invalid"
            id="undefinedMessage"
            tabIndex={-1}
          >
            What did you enter!?
          </div>
          <div
            className="below-message"
          >
            <div
              className="flex-row"
            >
              <span>
                Some content
              </span>
              <div
                className="cashflow-column cashflow-column-label"
              >
                <label
                  htmlFor="undefined_monthly"
                  id="undefinedLabel"
                  tabIndex="-1"
                >
                  Rent
                </label>
              </div>
            </div>
          </div>
        </div>
      </WithMessageAbove>
      <Surrounder
        Top={
          <InvalidMessage
            id="undefinedMessage"
            validRow={false}
          >
            What did you enter!?
          </InvalidMessage>
        }
      >
        <div
          className="surrounder"
        >
          <div
            className="top horizontal"
          >
            <InvalidMessage
              id="undefinedMessage"
              validRow={false}
            >
              <div
                className="invalid "
                id="undefinedMessage"
              >
                What did you enter!?
              </div>
            </InvalidMessage>
          </div>
          <div
            className="middle horizontal"
          >
            <div
              className="center horizontal content"
            >
              <span>
                Some content
              </span>
              <div
                className="cashflow-column cashflow-column-label"
              >
                <label
                  htmlFor="undefined_monthly"
                  id="undefinedLabel"
                  tabIndex="-1"
                >
                  Rent
                </label>
              </div>
            </div>
          </div>
        </div>
      </Surrounder>
    </div>
  </FormField>
</CashFlowRow>
`;

exports[`<CashFlowRow> renders with no message 1`] = `
<CashFlowRow
  label="Rent"
  validRow={true}
>
  <FormField
    className="cashflow"
    inline={true}
  >
    <div
      className="inline field cashflow"
    >
      <WithMessageAbove
        isUserError={false}
      >
        <div
          className="with-message-above "
        >
          <div
            className="below-message"
          >
            <div
              className="flex-row"
            >
              <span>
                Some content
              </span>
              <div
                className="cashflow-column cashflow-column-label"
              >
                <label
                  htmlFor="undefined_monthly"
                  id="undefinedLabel"
                  tabIndex="-1"
                >
                  Rent
                </label>
              </div>
            </div>
          </div>
        </div>
      </WithMessageAbove>
      <Surrounder
        Top={
          <span
            id="undefinedMessage"
          />
        }
      >
        <div
          className="surrounder"
        >
          <div
            className="top horizontal"
          >
            <span
              id="undefinedMessage"
            />
          </div>
          <div
            className="middle horizontal"
          >
            <div
              className="center horizontal content"
            >
              <span>
                Some content
              </span>
              <div
                className="cashflow-column cashflow-column-label"
              >
                <label
                  htmlFor="undefined_monthly"
                  id="undefinedLabel"
                  tabIndex="-1"
                >
                  Rent
                </label>
              </div>
            </div>
          </div>
        </div>
      </Surrounder>
    </div>
  </FormField>
</CashFlowRow>
`;
